modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoDataStoreTest.java

```diff
@@ -229,4 +229,47 @@ public void testSortBy() throws Exception {
         assertTrue(first.before(second));
         assertTrue(second.before(third));
     }
+
+    public void testIsNullFilter() throws Exception {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        PropertyName pn = ff.property("properties.nullableAttribute");
+        PropertyIsNull isNull = ff.isNull(pn);
+        Query q = new Query("ft1", isNull);
+        SimpleFeatureSource source = dataStore.getFeatureSource("ft1");
+        SimpleFeatureCollection features = source.getFeatures(q);
+        assertEquals(features.size(), 2);
+        SimpleFeatureIterator it = features.features();
+        while (it.hasNext()) {
+            SimpleFeature f = it.next();
+            assertNull(pn.evaluate(f));
+        }
+    }
+
+    public void testNotFilter() throws Exception {
```

Comment:
I would add a few extra tests:
- a not equal to between two properties
-  a not filter with and nested and filter with two branches
- an or with two branches containing a not filter with

Reply:
fixed